core/types/data_source.go

```diff
@@ -2,13 +2,16 @@ package types
 
 import (
 	"encoding/hex"
+	"errors"
 	"fmt"
 
 	"code.vegaprotocol.io/vega/libs/crypto"
 	"code.vegaprotocol.io/vega/libs/proto"
 	vegapb "code.vegaprotocol.io/vega/protos/vega"
 )
 
+var ErrMultipleSameKeyNamesInFilterList = errors.New("multiple keys with same name found in filter list")
```

Comment:
This error is duplicated in oracle spec I believe, we should probably use the same error there

Reply:
Yes, the problem here was that oracles package is already importing `types`, so here I will definitely not import oracles because I will create a cycle.
I can put the error only in this package and the oracle to import it.
But the error contains oracle logic domain.

Reply:
The oracles package is already importing the types package, and this error here is in the types package. So the error should only be defined here, and not in the oracles package. There's no issues with cyclical imports

Reply:
Yes, I placed it in the types package :+1: